<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QWidget" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>400</width>
    <height>750</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>400</width>
    <height>700</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Scientific Calculator</string>
  </property>
  <property name="styleSheet">
   <string notr="true">
    QWidget { background-color: rgb(30,30,30) }
    QPushButton {
        background-color: #2d2d2d;
        color: #ffffff;
        border: 1px solid #404040;
        border-radius: 12px;
        font-size: 16px;
        font-weight: bold;
        min-height: 56px;
    }
    QPushButton:hover { background-color: #3a3a3a; }
    QPushButton:pressed { background-color: #4a4a4a; }

    /* Function buttons (sin, cos, etc.) */
    QPushButton[class=&quot;func&quot;] {
        background-color: #1a1a1a;
        color: #00d4ff;
    }
    QPushButton[class=&quot;func&quot;]:hover { background-color: #2a2a2a; }

    /* Operator buttons */
    QPushButton[class=&quot;operator&quot;] {
        background-color: #ff9500;
        color: white;
    }
    QPushButton[class=&quot;operator&quot;]:hover { background-color: #ffb143; }

    /* Equals button */
    QPushButton#btnEquals {
        background-color: #00d4ff;
        color: black;
        font-weight: bold;
    }
    QPushButton#btnEquals:hover { background-color: #66e0ff; }

    /* Clear backspace */
    QPushButton#btnClear, QPushButton#btnBackspace {
        background-color: #ff3b30;
        color: white;
    }
    QPushButton#btnClear:hover, QPushButton#btnBackspace:hover {
        background-color: #ff6b66;
    }

    QLineEdit#display {
        background-color: #000000;
        color: #00ff00;
        font-size: 36px;
        font-family: &quot;Consolas&quot;, &quot;Courier New&quot;, monospace;
        padding: 20px;
        border: none;
        border-radius: 16px;
        text-align: right;
    }
    QLineEdit#history {
        background-color: #121212;
        color: #888888;
        font-size: 18px;
        padding: 8px 20px;
        border: none;
        border-radius: 12px;
        text-align: right;
    }
   </string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout_2">
   <property name="spacing">
    <number>0</number>
   </property>
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QVBoxLayout" name="verticalLayout">
     <property name="spacing">
      <number>12</number>
     </property>
     <property name="margin" stdset="0">
      <number>20</number>
     </property>
     <item>
      <widget class="QLineEdit" name="history">
       <property name="readOnly">
        <bool>true</bool>
       </property>
       <property name="placeholderText">
        <string/>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QLineEdit" name="display">
       <property name="text">
        <string>0</string>
       </property>
       <property name="alignment">
        <set>Qt::AlignmentFlag::AlignRight|Qt::AlignmentFlag::AlignTrailing|Qt::AlignmentFlag::AlignVCenter</set>
       </property>
       <property name="readOnly">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <layout class="QGridLayout" name="gridLayout">
       <property name="spacing">
        <number>10</number>
       </property>
       <item row="3" column="0">
        <widget class="QPushButton" name="btnPi">
         <property name="text">
          <string>π</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QPushButton" name="btn7">
         <property name="text">
          <string>7</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QPushButton" name="btnAcos">
         <property name="text">
          <string>cos⁻¹</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="7" column="6">
        <widget class="QPushButton" name="btnDot">
         <property name="text">
          <string>.</string>
         </property>
        </widget>
       </item>
       <item row="0" column="6">
        <widget class="QPushButton" name="btnDegRad">
         <property name="styleSheet">
          <string notr="true">background-color: #333333; color: #ff9500;</string>
         </property>
         <property name="text">
          <string>DEG</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="3" column="6">
        <widget class="QPushButton" name="btnSqrt">
         <property name="text">
          <string>√x</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="0" column="0">
        <widget class="QPushButton" name="btnClear">
         <property name="text">
          <string>C</string>
         </property>
         <property name="shortcut">
          <string>Ctrl+C</string>
         </property>
        </widget>
       </item>
       <item row="5" column="9">
        <widget class="QPushButton" name="btnSub">
         <property name="text">
          <string>−</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>operator</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="4" column="9">
        <widget class="QPushButton" name="btnMul">
         <property name="text">
          <string>×</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>operator</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="1" column="6">
        <widget class="QPushButton" name="btnLog">
         <property name="text">
          <string>log</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="1" column="9">
        <widget class="QPushButton" name="btnAlog">
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>10ⁿ</string>
         </property>
         <property name="class" stdset="0">
          <string>func</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2">
        <widget class="QPushButton" name="btnTan">
         <property name="text">
          <string>tan</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="0" column="2">
        <widget class="QPushButton" name="btnInv">
         <property name="styleSheet">
          <string notr="true">background-color: #333333; color: #ff9500;</string>
         </property>
         <property name="text">
          <string>Inv</string>
         </property>
         <property name="checkable">
          <bool>true</bool>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QPushButton" name="btn4">
         <property name="text">
          <string>4</string>
         </property>
        </widget>
       </item>
       <item row="4" column="2">
        <widget class="QPushButton" name="btn9">
         <property name="text">
          <string>9</string>
         </property>
        </widget>
       </item>
       <item row="3" column="9">
        <widget class="QPushButton" name="btnDiv">
         <property name="text">
          <string>÷</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>operator</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="4" column="6">
        <widget class="QPushButton" name="btnLeftParen">
         <property name="text">
          <string>(</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="7" column="2">
        <widget class="QPushButton" name="btnAns">
         <property name="text">
          <string>Ans</string>
         </property>
        </widget>
       </item>
       <item row="5" column="1">
        <widget class="QPushButton" name="btn5">
         <property name="text">
          <string>5</string>
         </property>
        </widget>
       </item>
       <item row="2" column="2">
        <widget class="QPushButton" name="btnAtan">
         <property name="text">
          <string>tan⁻¹</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="0" column="1">
        <widget class="QPushButton" name="btnBackspace">
         <property name="text">
          <string>⌫</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0">
        <widget class="QPushButton" name="btn1">
         <property name="text">
          <string>1</string>
         </property>
        </widget>
       </item>
       <item row="6" column="2">
        <widget class="QPushButton" name="btn3">
         <property name="text">
          <string>3</string>
         </property>
        </widget>
       </item>
       <item row="6" column="9">
        <widget class="QPushButton" name="btnAdd">
         <property name="text">
          <string>+</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>operator</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="6" column="1">
        <widget class="QPushButton" name="btn2">
         <property name="text">
          <string>2</string>
         </property>
        </widget>
       </item>
       <item row="7" column="0" colspan="2">
        <widget class="QPushButton" name="btn0">
         <property name="text">
          <string>0</string>
         </property>
        </widget>
       </item>
       <item row="6" column="6">
        <widget class="QPushButton" name="btnPercent">
         <property name="text">
          <string>%</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QPushButton" name="btnSin">
         <property name="text">
          <string>sin</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="7" column="9">
        <widget class="QPushButton" name="btnEquals">
         <property name="minimumSize">
          <size>
           <width>0</width>
           <height>58</height>
          </size>
         </property>
         <property name="styleSheet">
          <string notr="true"/>
         </property>
         <property name="text">
          <string>=</string>
         </property>
        </widget>
       </item>
       <item row="5" column="2">
        <widget class="QPushButton" name="btn6">
         <property name="text">
          <string>6</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QPushButton" name="btnCos">
         <property name="text">
          <string>cos</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="2" column="0">
        <widget class="QPushButton" name="btnAsin">
         <property name="text">
          <string>sin⁻¹</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QPushButton" name="btnExp">
         <property name="text">
          <string>eˣ</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QPushButton" name="btn8">
         <property name="text">
          <string>8</string>
         </property>
        </widget>
       </item>
       <item row="5" column="6">
        <widget class="QPushButton" name="btnRightParen">
         <property name="text">
          <string>)</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QPushButton" name="btnPow">
         <property name="text">
          <string>xʸ</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
       <item row="2" column="9">
        <widget class="QPushButton" name="btnFact">
         <property name="text">
          <string>!</string>
         </property>
         <property name="class" stdset="0">
          <string>func</string>
         </property>
        </widget>
       </item>
       <item row="2" column="6">
        <widget class="QPushButton" name="btnLn">
         <property name="text">
          <string>ln</string>
         </property>
         <property name="class" stdset="0">
          <stringlist>
           <string>func</string>
          </stringlist>
         </property>
        </widget>
       </item>
      </layout>
     </item>
    </layout>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections/>
</ui>
